lib/livebook/zta.ex

```diff
@@ -45,6 +45,16 @@ defmodule Livebook.ZTA do
   """
   @callback authenticate(name(), Plug.Conn.t(), keyword()) :: {Plug.Conn.t(), metadata() | nil}
 
+  @doc """
+  Logouts against the given name.
+  """
+  @callback logout(name(), Phoenix.LiveView.Socket.t()) :: :ok | :error | :unsupported
+
+  @doc """
+  Checks if the logout is supported against the given name.
+  """
+  @callback logout_supported?() :: boolean()
```

Comment:
Oh, I've just realised that since it's a behaviour we can do even better. Instead of `logout` and `logout_supported?` we can do `@optional_callbacks logout: 2`. And then on the caller we do:

```ex
logout_supported? = Code.ensure_loaded?(module) and function_exported?(module, :logout, 2)
```

Reply:
I like that way, I forgot about the optional callbacks! Nice